examples/hawkbit-example-mgmt-simulator/src/main/java/org/eclipse/hawkbit/mgmt/client/scenarios/ConfigurableScenario.java

```diff
@@ -278,19 +281,30 @@ private void runRollout(final MgmtDistributionSet set, final Scenario scenario)
         // start the created Rollout
         rolloutResource.start(rolloutResponseBody.getRolloutId());
 
-        waitUntilRolloutIsComplete(rolloutResponseBody.getRolloutId());
+        if (scenario.isWaitTillRolloutComplete()) {
+            waitUntilRolloutIsComplete(rolloutResponseBody.getRolloutId());
+        }
         LOGGER.info("Run rollout for set {} -> Done", set.getDsId());
     }
 
     private void waitUntilRolloutNoLongerExists(final Long id) {
+
+        HttpStatus httpStatus;
         do {
             try {
                 TimeUnit.SECONDS.sleep(5);
             } catch (final InterruptedException e) {
                 LOGGER.warn("Interrupted!");
                 Thread.currentThread().interrupt();
             }
-        } while (rolloutResource.getRollout(id).getStatusCode() != HttpStatus.NOT_FOUND);
+            try {
+                final ResponseEntity<MgmtRolloutResponseBody> rollout = rolloutResource.getRollout(id);
+                httpStatus = rollout.getStatusCode();
+            } catch (final FeignException e) {
+                LOGGER.info("FeignException " + e);
```

Comment:
I don't get the reasoning behind this change. Looks like a quick debug hack that has nothing to do with this pull request. Am I right?

Reply:
No, I ran into this issue when running the Scenario a second time and trying to delete an existing rollout. When we call `rolloutResource.getRollout(id)` and the rollout does not exist anymore , we will not get `ResponseEntity` with status code 404, but a feign exception is thrown, since we are using the feign client for our http request (exception are wrapped in feign exceptions). That's why we have to catch the feign exception and look if the reason was a 404.